ui/components/src/TimeChart/TimeChart.tsx

```diff
@@ -254,13 +267,38 @@ export const TimeChart = forwardRef<ChartInstance, TimeChartProps>(function Time
     __experimentalEChartsOptionsOverride,
     noDataVariant,
     timeZone,
+    tooltipPinnedCoords,
     isStackedBar,
   ]);
 
   return (
     <Box
       sx={{ height }}
       onClick={(e) => {
+        // Determine where on chart canvas to plot pinned crosshair as markLine.
+        const pointInGrid = getPointInGrid(e.nativeEvent.offsetX, e.nativeEvent.offsetY, chartRef.current);
+        if (pointInGrid === null) {
+          return;
+        }
+
+        // Clear previously set pinned crosshair
+        const isCrosshairPinned = seriesMapping[seriesMapping.length - 1]?.name === 'Pinned Crosshair';
```

Comment:
Sorry for the large PR, these ~15 lines is the majority of the pinned crosshair implementation. When I added this and disabled xAxis.axisPointer.snap, I realized we _really_ needed the datapoint hover state too to complete this feature (or else when the crosshair was pinned, you could not necessarily tell which datapoint on the left or right was actually showing in the tooltip).

Reply:
@foxbat07 also asked for datapoint symbols to have a hover state when close to your cursor

Reply:
> it became more clear that we really need the datapoint hover state to complete this feature

+1, can we keep xaxis snaping until we implement the data point hover state? otherwise it's hard to tell the relationship between the tooltip and the data point it correlates

cc @saminzadeh

Reply:
Well datapoint hover state is part of this PR! But maybe I need to tweak that so it’s more clear (symbols get larger) 🤔

For longer time ranges in seemed like hiding them was okay but could also show them more often, ideally they would show only on hover (that is end goal but was hoping this PR was adequate for now)

Reply:
ohh i was looking at last 6 hrs, so maybe that was why i didn't notice the hover state